sdk/storage/storage-blob/samples/typescript/basic.ts

```diff
@@ -12,15 +12,20 @@ import {
   SharedKeyCredential,
   TokenCredential,
   Models
-} from "../.."; // Change to "@azure/storage-blob" in your package
+} from "../../src"; // Change to "@azure/storage-blob" in your package
 
 async function main() {
   // Enter your storage account name and shared key
-  const account = "";
-  const accountKey = "";
+  // const account = "";
+  // const accountKey = "";
+
+  // Enter your connection string
+  const connectionString = "";
 
   // Use SharedKeyCredential with storage account and account key
-  const sharedKeyCredential = new SharedKeyCredential(account, accountKey);
+  // const sharedKeyCredential = new SharedKeyCredential(account, accountKey);
+  // Use SharedKeyCredential with connectionString
+  const sharedKeyCredential = new SharedKeyCredential(connectionString);
```

Comment:
I think we would want to extract url, account name, access key, suffix from the connection string. We won't change `SharedKeyCredential`.  Instead create an instance of it using the extracted info.  If constructor overload doesn't work we may do something like
```typescript
static function fromConnectionString(
  connectionString: string,
  options: BlobConnectionOptions
): ServiceURL {
  const { url, accountName, accountKey, ... } = extractPartsWithValidation(connectionString);
  const sharedKeyCredential = new SharedKeyCredential(accountName, accountKey);
  // using the new constructor that takes BlobConnectionOptions
  return new ServiceURL(url, {
    credential: credential,
    pipelineOptions: pipelineOptions
  });
}
```

Similar code need to added to all the XxxxURL types.